logistics_planning_base/models/logistics_schedule.py

```diff
@@ -0,0 +1,84 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models, fields
+
+TRANSPORT_TYPE = [
+    ('ground', 'Ground'),
+    ('ocean-going', 'Ocean-Going'),
+]
+
+class LogisticsSchedule(models.Model):
+    _name = 'logistics.schedule'
+    _description = 'logistics.schedule'
+
+    name = fields.Char(compute='_compute_name')
+    company_id = fields.Many2one('res.company', string='Company', default=lambda self: self.env.company)
+    state = fields.Selection(
+        selection=[
+            ('draft', 'Draft'),
+            ('ready', 'Ready'),
+            ('done', 'Done'),
+            ('cancel', 'Cancel'),
+        ],
+        default='draft'
+    )
+    type = fields.Selection(
+        selection=[
+            ('input', 'Input'),
+            ('output', 'Output'),
+        ]
+    )
+    stock_move_id = fields.Many2one('stock.move')
+    picking_id = fields.Many2one('stock.picking', related='stock_move_id.picking_id')
+    destination_partner_id = fields.Many2one('res.partner')
+    partner_id = fields.Many2one('res.partner')
+    transport_type = fields.Selection(
+        selection=TRANSPORT_TYPE
+    )
+    product_id = fields.Many2one('product.product')
+    product_uom = fields.Many2one('uom.uom')
+    product_uom_qty = fields.Float()
+    scheduled_load_date = fields.Date()
+    commitment_date = fields.Datetime()
+    logistic_price_unit_type = fields.Selection(
+        selection=[
+            ('trip', 'Trip'),
+            ('unit', 'Unit'),
+        ]
+    )
+    currency_id = fields.Many2one('res.currency', default=lambda self: self.env.company.currency_id)
+    logistic_price_unit = fields.Float(digits='Product Price')
+    logistic_price_unit_done = fields.Float(digits='Product Price')
+    carrier_id = fields.Many2one('res.partner')
+    license_plate_1 = fields.Char()
+    license_plate_2 = fields.Char()
+    note = fields.Text()
+
+    def _compute_name(self):
+        for record in self:
+            record.name = f'{record.picking_id.name} - {record.product_id.name}'
+
+    def action_logistics_schedule_form_view(self):
+        action = self.env.ref('logistics_planning_base.action_logistics_schedule_form')
+        action.res_id = self.id
+        return action.read()[0]
+
+    def action_logistics_schedule_cancel(self):
+        for record in self.env.context.get("active_ids"):
+            record = self.browse(record)
+            if record.state in ['draft', 'ready']:
+                record.state = 'cancel'
+                record.stock_move_id = False
```

Comment:
Simpler code IMO (not tested):
```suggestion
        self.browse(self.env.context.get("active_ids", [])).filtered(
            lambda x: x.state in ['draft', 'ready']
        ).write({"state": "cancel", "stock_move_id": False})
```